### ISSUE TITLE
Reply-To email header refers to "localhost"

### ISSUE THREAD

**Issue by AlphaJack** [fisharebest/webtrees#4018]
Hello, I just noticed that emails from webtrees have a SPAM reputation of 6/15, and that these six points come from the `Reply-To` header, which contains "localhost":

```email
Return-Path: <server@example.org>
Delivered-To: user@example.org
Received: from example.org
    by Jack-P2 with LMTP
    id r6kYA3vJK2FUEAAAUprYAg
    (envelope-from <server@example.org>)
    for <user@example.org>; Sun, 29 Aug 2021 19:52:59 +0200
X-Original-To: <user@example.org>
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=example.org;
    s=r; t=1630259577; h=from:from:reply-to:reply-to:subject:subject:date:date:
    message-id:message-id:to:to:cc:mime-version:mime-version:
    content-type:content-type; bh=aaa=;
    b=aaa=
Message-ID: <ee22a7be0077813bcd696d3924bd26d7@example.org>
Date: Sun, 29 Aug 2021 17:52:57 +0000
Subject: New registration at https://example.org/webtrees
From: webtrees <server@example.org>
Reply-To: webtrees <no-reply@localhost>
To: First Last <user@example.org>
MIME-Version: 1.0
Content-Type: multipart/alternative;
    boundary="_=_swift_aaa_=_"
X-Spam: Yes 
```

Is there an option to remove the header, or at least make the domain the same as the envelope one?  

Not related to the issue, but would it be possible to specify a custom name for the sender like "Webtrees | Example \<server@example.org\>" instead of "webtrees \<server@example.org\>"?

**Comment by jon48** [comment]
Regarding the last question, the sender name and sender address are set in the *Control Panel* > *Sending E-mail* page, where you can configure the "Sender name" and "Sender email".

As for the reply-to, as far as I can see, the 'no-reply@localhost' email is only used in specific cases when sending a message to the administrators (as part of the registration process), but it is indeed an hardcoded value, and there is no option to override it. Maybe that could be an additional setting in the *Sending e-mail* page.
Out of curiosity, what tool have you used to get this spam reputation score?

**Comment by AlphaJack** [comment]
> the sender name and sender address are set in the Control Panel > Sending E-mail page, where you can configure the "Sender name" and "Sender email"

Unfortunately I am unable to set it, there seems to be a client side validator for an email address (type="email" is set in HTML):
![Screenshot_20210830_203337](https://user-images.githubusercontent.com/19962243/131387797-208a88df-872e-4b3c-88ba-324f73525ccc.png)

> the 'no-reply@localhost' email is only used in specific cases when sending a message to the administrators

Good to know, I had two editors with a Protonmail address whose verification email was never sent to them (not even in SPAM, but postfix was SURE to have delivered the email correctly). My guess was a very high SPAM level, maybe there is something else in verification emails other than this header

> Out of curiosity, what tool have you used to get this spam reputation score?

Rspamd:
![Screenshot_20210830_202627](https://user-images.githubusercontent.com/19962243/131387263-0845da9d-966e-4341-a609-18ce0da63af6.png)

**Comment by fisharebest** [comment]
FYI, I use protonmail, and I receive these messages OK.  They don't go to my spam folder.

> Is there an option to remove the header, or at least make the domain the same as the envelope one?

We need a "no-reply" email address for things like password-reset requests, etc.
By definition, these need to be invalid - or route to `/dev/null`, etc.

> Not related to the issue, but would it be possible to specify a custom name for the sender like "Webtrees | Example <server@example.org>" instead of "webtrees <server@example.org>"?

This has already been implemented, and will be included in the 2.1.0 release.

**Comment by wkitty42** [comment]
On 8/30/21 2:42 PM, AlphaJack wrote:
>     the sender name and sender address are set in the Control Panel > Sending
>     E-mail page, where you can configure the "Sender name" and "Sender email"
> 
> Unfortunately I am unable to set it, there seems to be a client side validator 
> for an email address (type="email" is set in HTML):
> Screenshot_20210830_203337 
> <https://user-images.githubusercontent.com/19962243/131387797-208a88df-872e-4b3c-88ba-324f73525ccc.png>

ummm... sender names (left side of the '@') do not have spaces in them ;)

try myserver, my_server, my.server, or similar...

**Comment by jon48** [comment]
> Unfortunately I am unable to set it, there seems to be a client side validator for an email address (type="email" is set in HTML):

> ummm... sender names (left side of the '@') do not have spaces in them ;) try myserver, my_server, my.server, or similar...

I have probably caused the confusion: as Greg has mentioned, the separate "Sender name" / "Sender email" settings are available in the latest source code for 2.1.0-dev, but this has not been released yet, and 2.0.16 does not allow for the Sender display name to be modified (the setting only set the Sender email, and therefore needs to be a valid email).
